sycl/source/detail/async_alloc.cpp

```diff
@@ -95,24 +110,38 @@ __SYCL_EXPORT void *async_malloc_from_pool(sycl::handler &h, size_t size,
   auto &Adapter = h.getContextImplPtr()->getAdapter();
   auto &memPoolImpl = sycl::detail::getSyclObjImpl(pool);
 
-  // Get events to wait on.
-  auto depEvents = getUrEvents(h.impl->CGData.MEvents);
-  uint32_t numEvents = h.impl->CGData.MEvents.size();
+  // Get CG event dependencies for this allocation.
+  const auto &DepEvents = h.impl->CGData.MEvents;
+  auto UREvents = getUrEvents(DepEvents);
+  uint32_t numEvents = DepEvents.size();
```

Comment:
Think we only need this on line 143, i'd just define this variable next to that usage, or inline it.